### PR TITLE
Support for process types in SDL

### PR DESCRIPTION
**Pull request by rphh**
The argparse was replaced by custom argument parser.

### CODE REVIEW

File: sdl2promela/sdl2promela.py [hunk 1]
```diff
@@ -18,29 +16,125 @@
...
+    scl_files: [str]
+    """List of Stop Condition Language Files."""
+
+    def __init__(self):
+        self.output_filename = None
+        self.verbose = False
+        self.sdl_files = []
+        self.scl_files = []
+
+    def verify(self) -> bool:
```
Comment (Lurkerpas) [inline sdl2promela/sdl2promela.py:37]: Doc

File: sdl2promela/sdl2promela.py [hunk 2]
```diff
@@ -18,29 +16,125 @@
...
+        return True
+
+
+class ProgramOptionsParseException(Exception):
+    pass
```
Comment (Lurkerpas) [inline sdl2promela/sdl2promela.py:42]: Doc

File: sdl2promela/sdl/model.py [hunk 3]
```diff
@@ -964,16 +964,19 @@ class Model:
     """
 
     def __init__(self, process: ogAST.Process):
-        self.source = process
+        if process.instance_of_ref:
```
Comment (Lurkerpas) [inline sdl2promela/sdl/model.py:967]: Do we have any integration test to check this change?

Comment (Lurkerpas) [reply sdl2promela/sdl/model.py:969]: Yes we do now.

File: sdl2promela/sdl2promela.py (hunk 1 above)
Comment (rphh) [reply sdl2promela/sdl2promela.py:37]: Doc added, implementation changed (this was stub).

File: sdl2promela/sdl2promela.py (hunk 2 above)
Comment (rphh) [reply sdl2promela/sdl2promela.py:57]: Added.

**Review by Lurkerpas — APPROVED** [review]

**Pull request merged**